### PR TITLE
Autofuzz map support

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by kyakdan** [comment]
let's merge https://github.com/CodeIntelligenceTesting/jazzer/pull/230 first before reviewing this PR.

**Comment by fmeum** [comment]
@kyakdan I rebased the PR onto `main`.

File: agent/src/test/java/com/code_intelligence/jazzer/autofuzz/MetaTest.java [hunk 1]
```diff
@@ -121,6 +123,44 @@ public void testConsume() {
...
+            23, // remaining bytes
+            "value1",
+            (byte) 1, // do not return null for the string
+            27, // remaining bytes
+            "key2",
+            (byte) 0 // *do* return null for the string
+            ));
+  }
+
+  private Map<String, String> returnsStringStringMap() {
```
Comment (bertschneider) [inline agent/src/test/java/com/code_intelligence/jazzer/autofuzz/MetaTest.java:161]: Just a little nitpick: Should this method be moved up to the other test methods?

File: agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java [hunk 2]
```diff
@@ -344,6 +360,55 @@ static Object consume(FuzzedDataProvider data, Class<?> type, AutofuzzCodegenVis
                                     ", ", "new java.io.ByteArrayInputStream(new byte[]{", "})")));
       }
       return new ByteArrayInputStream(array);
+    } else if (type == Map.class) {
+      if (visitor != null) {
+        // Do not use Collectors.toMap() since it cannot handle null values.
+        visitor.pushGroup("java.util.stream.Stream.of(", ", ",
+            ").collect(java.util.HashMap::new, (map, e) -> map.put(e.getKey(), e.getValue()), java.util.HashMap::putAll)");
```
Comment (bertschneider) [inline agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java:367]: `Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue)` could be used here. I'm undecided which version I find more readable.

**Review by bertschneider — APPROVED** [review]
Really nice addition. 

I just added two comments/nitpicks, but feel free to merge as is.

File: agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java (hunk 2 above)
Comment (bertschneider) [reply agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java:367]: Ah, no, sorry. I saw the comment but forgot about it during the review, even though it's right there above the line.

**Review by bertschneider — APPROVED** [review]
Looks good. Thx for the clarifications.

**Pull request merged**